channeldb/migration32/mission_control_store.go

```diff
@@ -142,33 +145,49 @@ type paymentResultNew struct {
 	failure tlv.OptionalRecordT[tlv.TlvType3, paymentFailure]
 }
 
-// paymentFailure represents the presence of a payment failure. It may or may
-// not include additional information about said failure.
-type paymentFailure struct {
-	info tlv.OptionalRecordT[tlv.TlvType0, paymentFailureInfo]
-}
-
 // newPaymentFailure constructs a new paymentFailure struct. If the source
 // index is nil, then an empty paymentFailure is returned. This represents a
 // failure with unknown details. Otherwise, the index and failure message are
 // used to populate the info field of the paymentFailure.
 func newPaymentFailure(sourceIdx *int,
-	failureMsg lnwire.FailureMessage) *paymentFailure {
```

Comment:
as mentioned before I think we can also get rid of the sourceIdx pointer ?

Reply:
I'm currently not seeing a lot of advantages in removing the source index pointer and would like to keep further changes in this bug fix PR minimal (also with regards to changing the signature to return an error), since this is not touching the serialization. I can change it though if you feel strongly about it.

Reply:
ok cool let's keep it as is.